python/lsst/ts/ess/common/device/base_device.py

```diff
@@ -94,6 +94,19 @@ async def __aexit__(
     ) -> None:
         await self.close()
 
+    def __repr__(self) -> str:
+        return self.__str__()
+
+    def __str__(self) -> str:
+        vars_list = [
+            f"{var}={val!r}"
+            for var, val in vars(self).items()
+            if var not in ["log", "terminator"]
```

Comment:
I gently suggest a set or frozenset for the items to ignore, e.g. `if var not in {"log", "terminator"}`. It is allegedly more efficient, though should not make much difference either way in this case.